src/eval/Eval.ml

```diff
@@ -791,18 +786,25 @@ let create_cur_state_fields initcstate curcstate =
   (* Combine filtered list and curcstate *)
   pure (filtered_init @ curcstate)
 
+let check_contr libs_env cconstraint cfields initargs curargs =
+  let initargs' = CU.remove_noneval_args initargs in
+  let env = Env.bind_all libs_env initargs' in
+  let%bind () = eval_constraint cconstraint env in
+  let%bind field_vals = init_fields env cfields in
```

Comment:
Does this fix the problem of field initialisers being evaluated transition invocations, despite the results just being thrown away (once the contract is deployed, we just look up field values through IPC)?

This has been bugging me for ages.

Reply:
> Does this fix the problem of field initialisers being evaluated transition invocations, despite the results just being thrown away (once the contract is deployed, we just look up field values through IPC)?
> 
> This has been bugging me for ages.

Yes, this fixes that !.